src/FileCache.js

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) 2017 Electric Imp
+// This file is licensed under the MIT License
+// http://opensource.org/licenses/MIT
+
+'use strict';
+
+const fs = require('fs-extra');
+const url = require('url');
+const path = require('path');
+const minimatch = require('minimatch');
+const HttpReader = require('./Readers/HttpReader');
+const GithubReader = require('./Readers/GithubReader');
+const GITHUB_DIR = path.sep + 'github';
+const HTTP_DIR = path.sep + 'http';
+const DEFAULT_EXCLUDE_FILE_NAME = 'builder-cache.exclude';
+const CACHED_READERS = [GithubReader, HttpReader];
+const OUTDATE_DAYS = 1;
+
+class FileCache {
+
+  constructor(machine) {
+    this._useCache = false;
+    this._cacheDir = '.' + path.sep + 'cache';
+    this._excludeList = [];
+    this._machine = machine;
+    this._outdateTime = OUTDATE_DAYS * 86400000; // precalc milliseconds in one day
+  }
+
+  /**
+   * Transform url or github link to path and filename
+   * It is important, that path and filename are unique,
+   * because collision can break the build
+   * @param {string} link link to the file
+   * @return {string} folder and name, where cache file can be found
+   * @private
+   */
+ _getCachedPath(link) {
+    const ghRes = GithubReader.parseUrl(link);
```

Comment:
I think not, cose we don't cache local files, so we don't need cachedPath for them, and don't have to perform some of the cache logic for them